### PR TITLE
MONGOID-5173 Specs should use bang (!) methods as much as possible

### PR DESCRIPTION
**Pull request by johnnyshields**
Bang methods .create!, .save!, and .update_attributes! ensure that the methods actually persisted to the DB and do not have silent failures. They will raise an error (which will make test red) if they fail.

Follow-up from conversation in https://github.com/mongodb/mongoid/pull/5029

### CODE REVIEW

File: spec/mongoid/persistable/creatable_spec.rb [hunk 1]
```diff
@@ -4,14 +4,14 @@
 
 describe Mongoid::Persistable::Creatable do
 
-  describe ".create" do
+  describe ".create!" do
```
Comment (p-mongo) [inline spec/mongoid/persistable/creatable_spec.rb:7]: This removes all test coverage for `create` method.

**Review by p-mongo — COMMENTED** [review]
Hi Johnny,

Did you read (i.e. review) this diff? Green CI doesn't guarantee the tests are *correct* in that they are testing, for example, the desired functionality.

File: spec/mongoid/persistable/creatable_spec.rb (hunk 1 above)
Comment (johnnyshields) [reply spec/mongoid/persistable/creatable_spec.rb:7]: I'm still reading through the diffs I haven't done a final pass on this yet.

Comment (johnnyshields) [reply spec/mongoid/persistable/creatable_spec.rb]: OK, diff reviewed.

**Comment by johnnyshields** [comment]
@p-mongo in some tests, only one of `#create` or `#create!` is tested. For these, I have intentionally changed to `#create!`, because firstly, `#create!` calls `#create` internally (i.e. greater code coverage) and second, it is more likely to cause a failure (i.e. an error) if the model didn't save as the test writer intended.

In the process of making this PR I did find several places where models were not being saved as intended. The most common was cases of `Account` missing `name` field.

**Comment by johnnyshields** [comment]
@p-mongo this PR is going to continually generate conflicts as new PRs are merged so it would be ideal if we could merge it first.

**Comment by johnnyshields** [comment]
@p-mongo @comandeo if the specs are passing can we please merge this?

**Comment by johnnyshields** [comment]
@comandeo let's merge this?

File: spec/mongoid/persistable/savable_spec.rb [hunk 2]
```diff
@@ -296,7 +296,7 @@
           # no conflicts.
           #expect(truck.atomic_updates[:conflicts]).to eq nil
 
-          expect { truck.save! }.not_to raise_error
+          expect(truck.save).to eq true
```
Comment (p-mongo) [inline spec/mongoid/persistable/savable_spec.rb:299]: @johnnyshields I am having difficulty understanding the changes in this file, could you explain them please?

**Review by p-mongo — COMMENTED** [review]
The rest of this PR seems to only affect `update_attributes`. It looks like we currently only have tests for `update_attributes` and not `update_attributes!`, so perhaps both should be minimally covered?

File: spec/mongoid/persistable/savable_spec.rb (hunk 2 above)
Comment (johnnyshields) [reply spec/mongoid/persistable/savable_spec.rb:299]:
```
expect { truck.save! }.not_to raise_error
```

is equivalent to:

```
expect(truck.save).to eq true
```

Both statements verify the document saved successfully. I prefer the latter in this case; we are specifically testing that the document saved.

**Comment by johnnyshields** [comment]
@p-mongo is my comment here sufficient to merge? Tests are green...

**Review by p-mongo — APPROVED** [review]

**Review by comandeo — APPROVED** [review]

**Pull request merged**